File: benchmarks/java/set/Java_Persistent/SetDifferenceSmall.java
import de.hhu.stups.btypes.BSet;
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class SetDifferenceSmall {






    private BInteger counter;
    private BSet<BInteger> set;

    public SetDifferenceSmall() {
        counter = new BInteger(0);
        set = new BSet<>();
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000))).booleanValue()) {
            set = set.difference(new BSet<>(new BInteger(1)));
            counter = counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        SetDifferenceSmall exec = new SetDifferenceSmall();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));

        Runtime runtime = Runtime.getRuntime();
        long memory = runtime.totalMemory() - runtime.freeMemory();
        System.out.println("Used memory is bytes: " + memory);
        System.out.println("Used memory is megabytes: "
                + (memory / (1024L * 1024L)));
    }

}

File: benchmarks/cpp/set/SetDifferenceSmall.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BSet.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef SetDifferenceSmall_H
#define SetDifferenceSmall_H

using namespace std;

class SetDifferenceSmall {



    private:



        BInteger counter;
        BSet<BInteger > set;

    public:

        SetDifferenceSmall() {
            counter = (BInteger(0));
            set = (BSet<BInteger >());
        }

        void simulate() {
            while((counter.less((BInteger(5000000)))).booleanValue()) {
                set = set.difference((BSet<BInteger >((BInteger(1)))));
                counter = counter.plus((BInteger(1)));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    SetDifferenceSmall exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

